plugins/recover.c

```diff
@@ -10,6 +10,91 @@
 #include <plugins/libplugin.h>
 #include <unistd.h>
 
+static struct plugin *plugin;
+static struct plugin_timer *lost_state_timer;
+/* This tells if we are already in the process of recovery. */
+static bool recovery, already_has_peers;
+static void do_check_lost_peer (void *unused);
+static struct node_id local_id;
+
+static void entering_recovery_mode(void *unused) {
+	return;
+}
+
+static struct command_result *after_listpeerchannels(struct command *cmd,
+					             const char *buf,
+					             const jsmntok_t *params,
+					             void *cb_arg UNUSED)
+{
+	plugin_log(plugin, LOG_DBG, "Listpeerchannels called");
+	struct listpeers_channel **channels = json_to_listpeers_channels(tmpctx, buf, params);
+	for (size_t i=0; i<tal_count(channels); i++) {
+		struct listpeers_channel *chan = channels[i];
+		if (chan->lost_state) {
+			plugin_log(plugin, LOG_DBG, "Detected a channel with lost state, Entering Recovery mode!");
+			recovery = true;
```

Comment:
Parse json yourself here, rather than reusing json_to_listpeers_channels which will be going away...

Reply:
Done